### PR TITLE
To Do list: add & remove

### PR DESCRIPTION
**Pull request by devendra-alt**
# [To Do list: add & remove](https://github.com/microverseinc/curriculum-javascript/blob/main/todo-list/m3_add_remove.md)
## 👨‍💻 Features Added
✔️ Implemented Add new task function 
✔️ Added remove task function
✔️ Added update task function on the description
✔️ Created task object and stored in todoTask array
✔️ All tasks are stored in localstorage
✔️ Updated index of elements on the deletion of a task

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,37 +1,8 @@
...
-
-    todoListEl.appendChild(listItemEl);
-  });
-};
-
-RenderToDoList();
+window.addEventListener('load', () => {
+  RenderToDoList();
+  addTask();
+});
```
Comment (AlphaNtihinduka) [inline src/index.js]: - [x] Currently I can not be able to add the tasks. Kindly consider calling `RenderToDoList()` after `addTask()`. `RenderToDoList()` is responsible for rendering the tasks based on the `toDoTasks` array. When a new task is added using `addTask()`, it updates the `toDoTasks` array and then immediately calls `RenderToDoList()`. This ensures that the new task is included in the rendering process and displayed on the web page.
image

**Review by AlphaNtihinduka — CHANGES_REQUESTED** [review]
## Hi @devendra-alt 🙋‍♂️
Good job working on toDo list. You have done the design to do the expectation and incredibly well. 👏  

TO HIGHLIGHT:
- New JavaScript file created for the new functionality ✔️ 
- To-Do List saved in local storage ✔️ 
- Well-structured files ✅

# STATUS: Required Changes :recycle: 

close

## Required changes :recycle: 
- Kindly consider the inline comment ☺️.
- [x] Kindly ensure that you the `MIT` is not broken. The MIT license gives users :person_curly_hair:  express permission to reuse code for any purpose.

## Optional suggestions
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread or slack channel if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

HAPPY CODING AND GOOD LUCK :crossed_fingers: :four_leaf_clover:

File: src/tasks.js [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+      event.preventDefault();
+      const tmpObj = {
+        index: toDoTasks.length + 1,
+        completed: false,
+        description: taskInput.value,
+      };
+      toDoTasks.push(tmpObj);
+      localStorage.setItem('tasks', JSON.stringify(toDoTasks));
+      taskInput.value = '';
+      RenderToDoList();
```
Comment (wayungi) [inline src/tasks.js:90]: error
- [ ] This error appears when i try to add a task for the first first but once i have added the first item, it does not appear again. Please write checks against this condition 🤔

**Review by wayungi — CHANGES_REQUESTED** [review]
Hi @devendra-alt , 

image

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't been addressed yet.


 ## Highlights
- [ ] New JavaScript file created for the new functionality ✔️ 
- [ ] Well-structured files ✔️ 
- [ ]  Correct GitHub flow ✔️ 

 ## Required Changes ♻️
_Check the comments under the review._

- [ ] Please fix the broken Licence link. You can follow[ this link](https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/adding-a-license-to-a-repository) to learn how its done.The MIT license gives users person_curly_hair express permission to reuse code for any purpose. 🐰 


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by roshan-bajgain — APPROVED** [review]
Hi @devendra-alt, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- GitHub flow is followed. ✔️
- PR has a good title and summary. ✔️
- Good job on implementing the changes ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**